x-pack/plugin/core/src/main/java/org/elasticsearch/datastreams/logsdb/LogsdbIndexModeSettingsProvider.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.datastreams.logsdb;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.index.IndexSettingProvider;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+
+public class LogsdbIndexModeSettingsProvider implements IndexSettingProvider {
+    private static final Logger logger = LogManager.getLogger(LogsdbIndexModeSettingsProvider.class);
+    private static final PatternMatcher logsNameMatcher = PatternMatcher.forLogs();
+
+    public static final Setting<Boolean> CLUSTER_INDEX_MODE_LOGSDB_ENABLED = Setting.boolSetting(
+        "cluster.index.mode.logsdb.enabled",
```

Comment:
I wanted to use a different setting to avoid reusing an existing one. I think it would be better to remove the `cluster.logsdb.enabled` setting.

Reply:
What is wrong with the existing setting? Maybe just move it to here?

Reply:
I wanted to avoid having the same setting controlling two different things.

Reply:
The old mechanism doesn't work and I think we should remove it. Integrations can either hardcode index.mode=logsdb in index templates or they can rely on this index settings provider to enable logsdb.